crates/aggchain-proof-contracts/src/contracts.rs

```diff
@@ -0,0 +1,73 @@
+use aggchain_proof_core::Digest;
+use alloy::network::Ethereum;
+use alloy::primitives::B256;
+use alloy::sol;
+
+use crate::Error;
+
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[sol(rpc)]
+    GlobalExitRootManagerL2SovereignChain,
+    "contracts/global-exit-root-manager-l2-sovereign-chain.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[sol(rpc)]
+    PolygonZkevmBridgeV2,
+    "contracts/polygon-zkevm-bridge-v2.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[sol(rpc)]
+    PolygonRollupManager,
+    "contracts/polygon-rollup-manager.json"
+);
+
+sol!(
+    #[allow(missing_docs)]
+    #[allow(clippy::too_many_arguments)]
+    #[sol(rpc)]
+    AggchainFep,
+    "contracts/aggchain-fep.json"
+);
+
+pub(crate) type ZkevmBridgeRpcClient<RpcProvider> =
+    PolygonZkevmBridgeV2::PolygonZkevmBridgeV2Instance<(), RpcProvider, Ethereum>;
+
+pub(crate) type PolygonRollupManagerRpcClient<RpcProvider> =
+    PolygonRollupManager::PolygonRollupManagerInstance<(), RpcProvider, Ethereum>;
+
+pub(crate) type AggchainFepRpcClient<RpcProvider> =
+    AggchainFep::AggchainFepInstance<(), RpcProvider, Ethereum>;
+
+#[async_trait::async_trait]
+pub trait L2LocalExitRootFetcher {
+    async fn get_l2_local_exit_root(&self, block_number: u64) -> Result<Digest, Error>;
+}
+
+#[async_trait::async_trait]
+pub trait L2OutputAtBlockFetcher {
+    fn parse_l2_output_root(json: serde_json::Value) -> Result<L2OutputAtBlock, Error>;
```

Comment:
This function is related to the internal of the implementation, it shouldn't be on the trait I guess

Reply:
That was also my dilemma, I guess you are right, It shouldn't be on the trait.